src/wire_msg.rs

```diff
@@ -189,11 +259,15 @@ impl MsgHeader {
 
     fn from_bytes(bytes: [u8; MSG_HEADER_LEN]) -> Self {
         let version = u16::from_be_bytes([bytes[0], bytes[1]]);
-        let data_len = u32::from_be_bytes([bytes[2], bytes[3], bytes[4], bytes[5]]);
-        let usr_msg_flag = bytes[6];
+        let user_header_len = u32::from_be_bytes([bytes[2], bytes[3], bytes[4], bytes[5]]);
```

Comment:
My understanding that for `ECHO_SRVC_MSG_FLAG` msgs, the `header` & `dst` will be `Bytes::new`.
However, seems here we still trying to parse from a 4 bytes segment ?

Reply:
@maqi The lengths of the `header` and `dst` will always be in the `Msgheader`? In that case, these will both be `0x0000_0000`.